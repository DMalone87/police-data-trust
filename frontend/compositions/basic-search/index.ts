export { InputForm } from "./input-form"
export { ResultsTable } from "./results-table"
