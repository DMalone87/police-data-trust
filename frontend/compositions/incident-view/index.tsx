import IncidentViewHeader from "./incident-view-header"
import IncidentBody from "./incident-view-body"
import { Incident } from "../../helpers/incident"

export default function IncidentView(incident: Incident) {
  return (
    <>
      <IncidentViewHeader {...incident} />
      <IncidentBody {...incident} />
    </>
  )
}
