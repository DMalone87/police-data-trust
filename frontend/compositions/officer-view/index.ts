import OfficerHeader from "./officer-view-header"
import OptionalOfficerInfo from "./optional-officer-info"

export { OfficerHeader, OptionalOfficerInfo }
