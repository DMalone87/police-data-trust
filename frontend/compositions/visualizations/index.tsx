import Map from "./map"
import useData from "./map/useData"

export { Map, useData }
