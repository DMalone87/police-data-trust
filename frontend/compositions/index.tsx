import DashboardHeader from "./dashboard-header/dashboard-header"
import EnrollmentCallToAction from "./enrollment-call-to-action/enrollment-call-to-action"
import EnrollmentHeader from "./enrollment-header/enrollment-header"
import {
  PassportApplicationResponse,
  RegistrationResponse
} from "./enrollment-response/enrollment-response"
import PasswordAid from "./password-aid/password-aid"
import { Map } from "./visualizations"
import ProfileNav from "./profile-nav"
import ProfileInfo from "./profile-info"
import ProfileType from "./profile-type"
import EditProfileInfo from "./profile-edit"
import SavedResults from "./profile-saved-tables/saved-results"
import SavedSearches from "./profile-saved-tables/saved-searches"
import LandingPage from "./landingpage/landing"

export {
  DashboardHeader,
  EnrollmentCallToAction,
  EnrollmentHeader,
  PassportApplicationResponse,
  PasswordAid,
  RegistrationResponse,
  Map,
  ProfileNav,
  ProfileInfo,
  EditProfileInfo,
  ProfileType,
  SavedResults,
  SavedSearches,
  LandingPage
}
