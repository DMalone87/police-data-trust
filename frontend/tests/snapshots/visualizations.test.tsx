import * as React from "react"
import { Map } from "../../compositions"
import { render } from "../test-utils"

it("renders Map correctly", async () => {
  const { container, findByTitle } = render(<Map />)
  await expect(findByTitle("US Map Graphic")).resolves.toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
