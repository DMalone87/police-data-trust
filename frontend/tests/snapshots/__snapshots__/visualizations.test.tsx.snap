// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Map correctly 1`] = `
<div>
  <div
    class="mapContainer"
    id="map-container"
  >
    <div
      class="mapWrapper"
      id="map-wrapper"
    >
      <svg
        aria-labelledby="title desc"
        class="mapData"
        id="map-svg"
        role="img"
        viewBox="0, 0, 1200, 700"
      >
        <title>
          US Map Graphic
        </title>
        <desc>
          A data graphic showing incidents of police violence by state.
        </desc>
        <g
          id="zoom-container"
        >
          <svg
            id="map"
            viewBox="0, 0, 1200, 700"
          >
            <rect
              fill="yellow"
              height="100"
              width="100"
              x="100"
              y="100"
            />
          </svg>
          <svg
            class="markerLayer"
            height="100%"
            id="marker-layer"
            viewBox="0, 0, 1200, 700"
            width="100%"
          />
        </g>
      </svg>
      <div
        aria-describedby="nullDetails"
        aria-label="Detailed information about null"
        class="popUp popUpClosed"
        id="popup"
        role="tooltip"
        style="transform: translate(undefinedpx, undefinedpx) translateX(50%);"
      >
        <p
          class="popUpHead"
          id="head"
        />
        <div
          class="popUpBody"
          id="body"
        />
      </div>
      <div
        style="position: absolute; right: 0px; bottom: 0px;"
      >
        <div
          class="chartKeyContainer"
          id="map-key"
        >
          <div
            class="KeyEntry"
          >
            <div />
            <div
              class="keyGradientEntry"
            >
              <label
                class="label"
              >
                minimum value
              </label>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(124, 174, 215)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(105, 144, 186)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(86, 113, 157)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(67, 83, 128)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(48, 52, 99)"
                  height="20"
                  width="20"
                />
              </svg>
              <label
                class="label"
              >
                maximum value
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mapButton"
      >
        <button
          class="primaryButton"
          id="zoom-reset"
        >
          Reset Zoom
        </button>
      </div>
    </div>
  </div>
</div>
`;
